site/en/tutorials/structured_data/imbalanced_data.ipynb

```diff
@@ -458,7 +506,7 @@
         "Now create and train your model using the function that was defined earlier. Notice that the model is fit using a larger than default batch size of 2048, this is important to ensure that each batch has a decent chance of containing a few positive samples. If the batch size was too small, they would likely have no fraudulent transactions to learn from.\n",
```

Comment:
Even though they're in a way equivalent, this line is a clue about why resampling if more effective than reweighting.

- reweighting you see a few important examples in each batch
- resampling you see a much wider distribution each batch.

Reply:
Thanks, I really learned something new by you pointing this out.

Reply:
Do you know any canonical literatur on this subject? How Ccass imbalance effects minibatch gradient descent?

Reply:
I'm having trouble finding a reference. But it seems clear that it would reduce the gradient variance.

Reply:
In case you find a reference, I'm really very very interested because the whole "imbalanced class problem" eludes me and it is rarely stated what the actual problem is and, e.g., why on should ever use class weights.
https://doi.org/10.1214/14-AOS1220, for instance, focuses on computational costs.
My current understanding is that "the problem" is mainly a computational one, either CPU/memory efficiency or solver/optimization related, but not a statistical problem (except a small effective sample size).